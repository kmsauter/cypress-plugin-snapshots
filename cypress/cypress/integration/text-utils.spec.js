const { cloneDeep } = require('lodash');
const { runSuites, dataRaw, dataFormatted } = require('../fixtures/test-data');
const { isHtml, getSubject } = require('../../../src/utils/commands/index');

describe('text - utils', () => {
  runSuites('isHtml', function (item, itemName, suiteName) {
    expect(isHtml(item)).to.equal(suiteName == 'html');
  });

  runSuites('getSubject', function (item, itemName, suiteName) {
    expect(getSubject(item)).to.equal(dataFormatted[suiteName][itemName]);
  });

  describe('normalizeObject', () => {
    it('true', () => {
      cy.wrap(dataRaw.json.object).toMatchSnapshot({ normalizeJson: true });
    });

    /* bug #66 */
    it.skip('false', () => {
      cy.wrap(dataRaw.json.object).toMatchSnapshot({ normalizeJson: false });
    });
  });

  describe('removeExcludedFields', () => {
    it('removeExcludedFields', () => {
      const obj = {
        a: 1,
        ignore: 2,
        b: 3,
        ignore: 4,
        c: {
          a: 5,
          ignore: 6
        },
        d: [1, { ignore: 'a' }]
      };
      cy.wrap(obj).toMatchSnapshot();
    });
  });

  /* bug #32 */
  describe.skip('keepKeysFromExpected', ()=> {
    it('ignoreExtraFields: true, ignoreExtraArrayItems: false', () => {
      let obj = cloneDeep(dataRaw.json.object);
      obj.extraField = 'extraField';
      obj.nested.extraField = 'extraField';
      obj.array[4].extraField = 'extraField';
      obj.nested.array[6].extraField = 'extraField';

      cy.wrap(obj).toMatchSnapshot({
        ignoreExtraFields: true,
        ignoreExtraArrayItems: false
      });
    });

    it('ignoreExtraFields: false, ignoreExtraArrayItems: true', () => {
      let obj = cloneDeep(dataRaw.json.object);
      obj.array.push('extraItem');
      obj.nested.array.push('extraItem');

      cy.wrap(obj).toMatchSnapshot({
        ignoreExtraFields: false,
        ignoreExtraArrayItems: true
      });
    });

    it('ignoreExtraFields: true, ignoreExtraArrayItems: true', () => {
      let obj = cloneDeep(dataRaw.json.object);
      obj.extraField = 'extraField';
      obj.nested.extraField = 'extraField';
      obj.array[4].extraField = 'extraField';
      obj.nested.array[6].extraField = 'extraField';
      obj.array.push('extraItem');
      obj.nested.array.push('extraItem');

      cy.wrap(obj).toMatchSnapshot({
        ignoreExtraFields: true,
        ignoreExtraArrayItems: true
      });
    });
  });
});
