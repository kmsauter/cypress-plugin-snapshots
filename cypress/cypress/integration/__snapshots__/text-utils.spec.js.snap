exports[`text - utils > normalizeObject > true #0`] =
{
  "array": [
    "string",
    5,
    true,
    null,
    {
      "a": 1,
      "b": 2
    },
    null
  ],
  "date": "2019-02-11T05:00:00.000Z",
  "false": false,
  "nested": {
    "array": [
      2,
      "string",
      true,
      null,
      null,
      "2019-02-11T05:00:00.000Z",
      {
        "a": 1,
        "b": 2
      }
    ],
    "number": 1,
    "true": true
  },
  "null": null,
  "string": "string"
};

exports[`text - utils > normalizeObject > false #0`] =
{
  "false": false,
  "string": "string",
  "null": null,
  "array": [
    "string",
    5,
    true,
    null,
    {
      "b": 2,
      "a": 1
    },
    null
  ],
  "date": "2019-02-11T05:00:00.000Z",
  "nested": {
    "array": [
      2,
      "string",
      true,
      null,
      null,
      "2019-02-11T05:00:00.000Z",
      {
        "b": 2,
        "a": 1
      }
    ],
    "true": true,
    "number": 1
  }
};

exports[`text - utils > removeExcludedFields > removeExcludedFields #0`] =
{
  "a": 1,
  "b": 3,
  "c": {
    "a": 5
  },
  "d": [
    1,
    {}
  ]
};

exports[`text - utils > keepKeysFromExpected > ignoreExtraFields: true, ignoreExtraArrayItems: false #0`] =
{
  "array": [
    "string",
    5,
    true,
    null,
    {
      "a": 1,
      "b": 2
    },
    null
  ],
  "date": "2019-02-11T05:00:00.000Z",
  "false": false,
  "nested": {
    "array": [
      2,
      "string",
      true,
      null,
      null,
      "2019-02-11T05:00:00.000Z",
      {
        "a": 1,
        "b": 2
      }
    ],
    "number": 1,
    "true": true
  },
  "null": null,
  "string": "string"
};

exports[`text - utils > keepKeysFromExpected > ignoreExtraFields: false, ignoreExtraArrayItems: true #0`] =
{
  "array": [
    "string",
    5,
    true,
    null,
    {
      "a": 1,
      "b": 2
    },
    null
  ],
  "date": "2019-02-11T05:00:00.000Z",
  "false": false,
  "nested": {
    "array": [
      2,
      "string",
      true,
      null,
      null,
      "2019-02-11T05:00:00.000Z",
      {
        "a": 1,
        "b": 2
      }
    ],
    "number": 1,
    "true": true
  },
  "null": null,
  "string": "string"
};

exports[`text - utils > keepKeysFromExpected > ignoreExtraFields: true, ignoreExtraArrayItems: true #0`] =
{
  "array": [
    "string",
    5,
    true,
    null,
    {
      "a": 1,
      "b": 2
    },
    null
  ],
  "date": "2019-02-11T05:00:00.000Z",
  "false": false,
  "nested": {
    "array": [
      2,
      "string",
      true,
      null,
      null,
      "2019-02-11T05:00:00.000Z",
      {
        "a": 1,
        "b": 2
      }
    ],
    "number": 1,
    "true": true
  },
  "null": null,
  "string": "string"
};
